File: AtCoder/447A-QAQ.go
package main

import (
	"fmt"
)

func main() {
	var s string
	fmt.Scan(&s)

	n := len(s)
	count := 0

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			for k := j + 1; k < n; k++ {
				if s[i] == 'Q' && s[j] == 'A' && s[k] == 'Q' {
					count++
				}
			}
		}
	}

	fmt.Println(count)
}

File: AtCoder/447A-QAQ.rs
use std::io::{self, BufRead};

fn main() {
  let stdin = io::stdin();
  let mut iterator = stdin.lock().lines();

  let s = iterator.next().unwrap().unwrap();
  let n = s.len();

  let chars: Vec<char> = s.chars().collect();
  let mut count = 0;

  for i in 0..n {
    for j in (i + 1)..n {
      for k in (j + 1)..n {
        if chars[i] == 'Q' && chars[j] == 'A' && chars[k] == 'Q' {
          count += 1;
        }
      }
    }
  }

  println!("{}", count);
}
